system/Database/BaseConnection.php

```diff
@@ -302,6 +309,10 @@ public function __construct(array $params)
 		{
 			$this->$key = $value;
 		}
+
+		$queryClass = str_replace('Connection', 'Query', get_class($this));
+
+		class_exists($queryClass) && $this->queryClass = $queryClass;
```

Comment:
While short circuiting is not wrong, these instances may pose problems with static analysis tools on undiscovered symbols. I suggest using the braces here:
```php
if (class_exists($queryClass))
{
    $this->queryClass = $queryClass;
}
```